src/v/kafka/server/tests/fetch_bench.cc

```diff
@@ -247,34 +278,46 @@ struct fetch_bench_fixture : redpanda_thread_fixture {
 using large_fetch_t = fetch_bench_fixture<large_fetch_config>;
 using small_fetch_t = fetch_bench_fixture<small_fetch_config>;
 
+fetch_request_config single_partition_req_config(model::topic t) {
+    return fetch_request_config{fetch_topic{
+      .name = std::move(t),
+      .partitions = {fetch_part{model::partition_id(0), model::offset(0), 1}}}};
+}
+
+fetch_request_config multi_partition_req_config(model::topic t) {
+    return fetch_request_config{fetch_topic{
+      .name = std::move(t),
+      .partitions = {
+        fetch_part{model::partition_id(0), model::offset(0), 1},
+        fetch_part{model::partition_id(1), model::offset(0), 1}}}};
+}
+
 PERF_TEST_CN(large_fetch_t, single_partition_fetch) {
     static model::topic t = co_await initialize_single_partition_topic();
 
-    co_return co_await fetch_from(
-      t, {fetch_part{model::partition_id(0), model::offset(0), 1}});
+    co_return co_await fetch_from_local_shard(single_partition_req_config(t));
 }
 
 PERF_TEST_CN(small_fetch_t, single_partition_fetch) {
     static model::topic t = co_await initialize_single_partition_topic();
 
-    co_return co_await fetch_from(
-      t, {fetch_part{model::partition_id(0), model::offset(0), 1}});
+    co_return co_await fetch_from_local_shard(single_partition_req_config(t));
 }
 
 PERF_TEST_CN(large_fetch_t, multi_partition_fetch) {
     static model::topic t = co_await initialize_multi_partition_topic();
 
-    co_return co_await fetch_from(
-      t,
-      {fetch_part{model::partition_id(0), model::offset(0), 1},
-       fetch_part{model::partition_id(1), model::offset(0), 1}});
+    // One partition will be from the same shard, while the other will be from a
+    // foreign shard. This will hopefully allow us to detect if more or less
+    // cross shard calls are being made for a fetch.
+    co_return co_await fetch_from_local_shard(multi_partition_req_config(t));
```

Comment:
In the future it might be even better to just make these separate benchmarks: local and remote fetch, rather than mixing the two together in a 1:1 ratio as it will help clearly separate out changes which affect one flow more than another.

Reply:
There is the single partition fetch test case which will always fetch from the local shard only. For the multi-partition case it'd be rather easy to add a case where we're fetching from two partitions on the local shard. I'll add it in if you think its worthwhile.